### DISCUSSION TITLE
VSCode suggestions does not work inside Qtile config file

### DISCUSSION THREAD

**Discussion by Scotty78211** [qtile/qtile#3571] [Q&A]
I have seen some people have smart suggestions while they edit their config file in vscode, for example when they create a key bindings they get to choice one of the Lazy functions such as `lazy.spawn()`. instead of keep referring to the documentation, they just hover over the object or press `ctrl+space` and vscode magically return information about that object and gives suggestions to select from.

In my case this is not working, tried installing Python extension and used venv but no luck.
![vscode_qtile_debugging_not_working](https://user-images.githubusercontent.com/76183029/169479310-7bb49b00-4681-4e62-babc-cdec28311257.png)

**Comment by elParaguayo** [comment]
I don't know how the smart suggestions work in vscode but I can see a few issues:
1. vscode may struggle to understand what the `qtile` object represents. It's default value is `None`:
https://github.com/qtile/qtile/blob/2a0b91799ceccb23a8d67b85ce48395d9a9d5e2a/libqtile/__init__.py#L23

2. `lazy` objects are special in that the objects and methods are lazily evaluated (e.g. at the time you try to run the binding) and the command names are based on parsing of methods in various classes. I doubt the vscode suggestion code is able to work this out on the fly.
